src/lib/tax-brackets.ts

```diff
@@ -4,11 +4,72 @@ import { TERRITORIES } from '../data/types/states';
 import { TaxEstimate } from '../data/types/tax';
 import { roundCents } from './rounding';
 
+// This basically works like deductions in federal taxes. Everyone is allowed to
+// claim a "personal exemption", which deducts this amount from taxable income.
+// https://www.mass.gov/info-details/massachusetts-personal-income-tax-exemptions
+const MA_EXEMPTIONS: { [S in FilingStatus]: number } = {
+  [FilingStatus.Single]: 4400,
+  [FilingStatus.MarriedFilingSeparately]: 4400,
+  [FilingStatus.HoH]: 6800,
+  [FilingStatus.Joint]: 8800,
+  [FilingStatus.QualifyingWidower]: 8800,
+};
+
+// Income over this amount is subject to the 4% surtax (a.k.a. "millionaire's
+// tax"). The threshold goes up every year, which is why it's not an even $1M.
+// This is for the 2025 tax year.
+// https://www.mass.gov/info-details/4-surtax-on-taxable-income-over-1000000
+const MA_SURTAX_THRESHOLD = 1_083_150;
+
+// Source: last page of
+// https://www.tax.newmexico.gov/individuals/wp-content/uploads/sites/5/2021/12/PIT-rates_2005_2021.pdf
+const NM_BRACKETS: {
+  [S in FilingStatus]: { maxIncome: number; rate: number }[];
+} = {
+  [FilingStatus.Single]: [
+    { maxIncome: 5500, rate: 0.017 },
+    { maxIncome: 11000, rate: 0.032 },
+    { maxIncome: 16000, rate: 0.047 },
+    { maxIncome: 210000, rate: 0.049 },
+    { maxIncome: Infinity, rate: 0.059 },
+  ],
+  [FilingStatus.MarriedFilingSeparately]: [
+    { maxIncome: 4000, rate: 0.017 },
+    { maxIncome: 8000, rate: 0.032 },
+    { maxIncome: 12000, rate: 0.047 },
+    { maxIncome: 157500, rate: 0.049 },
+    { maxIncome: Infinity, rate: 0.059 },
+  ],
+  // Below three all the same
```

Comment:
would it make sense to turn the arrays into a variable and use that three times?